packages/react-virtual/src/index.tsx

```diff
@@ -43,6 +43,10 @@ function useVirtualizerBase<
   instance.setOptions(resolvedOptions)
 
   React.useEffect(() => {
+    if (!instance.scrollElement) {
+      instance._willUpdate()
+    }
+
```

Comment:
I think better would be to change the `useEffect` to `useIsomorphicLayoutEffect`
```suggestion
  useIsomorphicLayoutEffect(() => {
```

then order will be correct 
```
instance._didMount, cleanup
instance._willUpdate, cleanup
instance._didMount
instance._willUpdate
```

Reply:
Hmm, interesting, was there a reason `useEffect` was used instead of  `useIsomorphicLayoutEffect` since the [beginning](https://github.com/TanStack/virtual/commit/4f38da468e5db4f7760d2c7845bc7d624fdcf40d#diff-d7f76e6511ee16ef9493c403e9f81f738e6d281dd9490890be946a9811f73d7aR39)?

Reply:
Overall when possible useEffect should preferred, there was no clear reason till now ot use  `useIsomorphicLayoutEffect` imho. 

> useLayoutEffect is a version of [useEffect](https://react.dev/reference/react/useEffect) that fires before the browser repaints the screen.

As this is called in limited way imho changing it to layout effect should be fine.

Reply:
@hophiphip please update the PR, then we can merge it! Thanks for looking into this.

Reply:
@piecyk, updated PR